src/main.rs

```diff
@@ -28,43 +28,59 @@ async fn main() -> anyhow::Result<()> {
     let cfg = proj_dirs.config_dir().join("websocket-token");
 
     let exists = tokio::fs::try_exists(&cfg).await;
-    if !exists.unwrap_or(false) {
-        anyhow::bail!(
-            "\
-No OBS WebSocket authentication token found.
-Open OBS, go to Tools -> WebSocket Server Settings, \
-hit 'Show Connect Info', and place that token in the file {}\
-            ",
-            cfg.display()
-        );
-    }
-    let pw = tokio::fs::read_to_string(&cfg)
-        .await
-        .context("read OBS WebSocket autheticator from obs-do config file")?;
 
-    let client = if let Ok(client) = Client::connect("localhost", 4455, Some(pw.trim())).await {
-        let version = client
-            .general()
-            .version()
-            .await
-            .context("get OBS version")?;
-        eprintln!(
-            "Connected to OBS: {} / {}",
-            version.obs_version, version.obs_web_socket_version
-        );
-        client
-    } else {
-        anyhow::bail!(
-            "\
+    let pw = match exists {
+        Ok(exists_bool) => {
+            if exists_bool {
+                Some(
+                    tokio::fs::read_to_string(&cfg)
+                        .await
+                        .unwrap()
+                        .trim()
+                        .to_string(),
+                )
+            } else {
+                eprintln!("Attempting to connect to OBS in password-less mode.");
+                None
+            }
+        }
+        Err(e) => {
+            anyhow::bail!("Error: {:?}", e);
+        }
```

Comment:
Here's a slightly nicer to read version :)
```suggestion
        Ok(true) => {
            Some(
                tokio::fs::read_to_string(&cfg)
                    .await
                    .unwrap()
                    .trim()
                    .to_string(),
            )
        Ok(false) => {
            eprintln!("Attempting to connect to OBS in password-less mode.");
            None
        }
        Err(e) => {
            anyhow::bail!("Failed to read OBS WebSocket password file {}: {e:?}", cfg.display());
        }
```

Reply:
I went ahead and merged this, but you may find it useful as an example!